airflow/jobs/job.py

```diff
@@ -123,6 +123,10 @@ def __init__(self, executor=None, heartrate=None, **kwargs):
     def executor(self):
         return ExecutorLoader.get_default_executor()
 
+    @cached_property
```

Comment:
This means the field gets recauclated for each Job instance. Is this good enough? Should we cache this class-wide instead?

Reply:
As @dirrao pointed out, this is already how we do it for the executor property for a while now, since neither is on the ORM model. Jobs aren't created that often and the ExecutorLoader caches all executors, so the cost is already paid and it is a quick lookup here.

Reply:
If this is cached elsewhere anyway, I guess my main issue is it is unclear why this property is more useful than just calling the underlying method directly.

Reply:
We need to ensure that an init of all executors is called at some point and then store that on a variable somewhere for repeated access within each job. This is simply a tidy/convenient spot to do that, since multiple jobs need access to these executors (scheduler and backfill at the least). So it's nice to do it once here, and then reference it from `self.job.executors` within the various JobRunners after instead of calling directly and repeating things in each Runner.